Cabal/src/Distribution/Simple/GHC/Internal.hs

```diff
@@ -377,6 +378,18 @@ includePaths lbi bi clbi odir =
          | dir <- mapMaybe (symbolicPathRelative_maybe . unsafeCoerceSymbolicPath) $ includeDirs bi
          ]
 
+allGhcOptExtra :: BuildInfo -> [String]
+allGhcOptExtra bi =
+  ["-optP" ++ opt | opt <- cppOptions bi]
+    ++ ["-opta" ++ opt | opt <- asmOptions bi]
+    ++ ["-optJSP" ++ opt | opt <- jsppOptions bi]
+    ++ ["-optl" ++ opt | opt <- ldOptions bi]
+    ++ ["-optc" ++ opt | opt <- ccOptions bi]
+#if __GLASGOW_HASKELL__ >= 810
```

Comment:
this is wrong. You should check for the configured compiler, not the one used to compile `Cabal`.